pytest.ini

```diff
@@ -9,25 +9,36 @@ filterwarnings=
 	# Fail on warnings
 	error
 
+	# Workarounds for pypa/setuptools#3810
+	# Can't use EncodingWarning as it doesn't exist on Python 3.9.
+	# These warnings only appear on Python 3.10+
+	default:'encoding' argument not specified
+	# Until Python 3.9 support is dropped, we can't use `encoding="locale"` directly,
+	# but we'd also prefer switching to `encoding="utf-8"` wherever possible
+	# in preparation of https://peps.python.org/pep-0686/
+	# In the mean time, it causes too much noise in tests
+	ignore:'encoding' argument not specified::setuptools
+	ignore:'encoding' argument not specified::pkg_resources
```

Comment:
I think this might be too generic and that would avoid identifying if new instances of the warning are introduced.

Would it be too complicated to refine this, now that we improved a little bit the situation? Is it worthy? (This is fundamentally a question/brainstorm, not necessarily prompting any change).

Reply:
Looking at https://github.com/pypa/setuptools/pull/4234/files, there's only 2 relevant locations left in `pkg_resources`

Reply:
Looks like there's 22 `EncodingWarning` left (down from the initial 124)
- 2 are in pip and happen outside of pytest
- 1 is `UTF-8 Mode affects locale.getpreferredencoding(). Consider locale.getencoding() instead.`, which this PR fixes (but could be split into its own small PR)
- 5 are from upstream/external libraries
- 2 are form _distutils


That leaves 12 `EncodingWarning` in tests raised from the following setuptools files
```ini
	# TODO: we should try to remove these lines gradually
	ignore:'encoding' argument not specified::setuptools._distutils.text_file
	ignore:'encoding' argument not specified::setuptools._imp
	ignore:'encoding' argument not specified::setuptools.command.bdist_egg
	ignore:'encoding' argument not specified::setuptools.command.easy_install
	ignore:'encoding' argument not specified::setuptools.command.install_scripts
	ignore:'encoding' argument not specified::setuptools.command.setopt
	ignore:'encoding' argument not specified::setuptools.dist
	ignore:'encoding' argument not specified::setuptools.package_index
	ignore:'encoding' argument not specified::setuptools.sandbox
	ignore:'encoding' argument not specified::setuptools.wheel
```